app/assets/javascripts/discourse/templates/components/user-card-contents.hbs

```diff
@@ -92,6 +96,21 @@
       tagName=""}}
     </div>
 
+    {{#if loading}}
```

Comment:
I am wondering, wouldn't it be easier to insert at line 3 something like:

```handlebars
{{#if loading}}
  <div class="card-row first-row">
    <div class="user-card-avatar">
      <div class="card-avatar-placeholder animated-placeholder"></div>
    </div>
  </div>

  <div class="card-row second-row">
    <div class="animated-placeholder"></div>
  </div>
  <div class="card-row third-row">
    <div class="animated-placeholder"></div>
  </div>
  <div class="card-row fourth-row">
    <div class="animated-placeholder"></div>
  </div>
  <div class="card-row sixth-row">
    <div class="animated-placeholder"></div>
  </div>
{{else}}
  {{!-- current code --}}
{{/if}}
```

This way you will not need the changes to discourse-cakeday and who knows how many other private plugins.

Also, I believe this will fix other plugin connectors that might break because are not in a `{{if loading}}...{{/if}}` block.

Reply:
Pushed this approach because I think it is more clean than multiple `#if loading` in between lines but that still didn't fix the error with `discourse-cakeday` because its test instantly tries to find a class on the card as soon as it appears and then gets nothing because the user hasn't finished loading yet.

Reply:
Ah, I understand. Had another look at the way `discourse-cakeday` tests this and I see that the check is wrapped under a `andThen`. It should work, which makes me think that the promise chain is broken at some point (which makes `andThen` proceed before the card is actually shown).

Reply:
No the chain is correct but when card is shown it only shows the loading animation until the user is loaded but the `find()` already takes place. It worked before because the card only showed after the user has been fully loaded.